src/store/website-status-context.tsx

```diff
@@ -30,8 +33,20 @@ export const WebsiteStatusProvider = ({ children }: WebsiteStatusProviderProps)
     })
 
     useEffect(() => {
-        send()
-    }, [send])
+        if (wsData) {
+            setData(wsData)
+            statusDataPromiseResolve(wsData)
+        } else if (!statusDataPromise) {
+            statusDataPromise = new Promise((resolve) => {
+                statusDataPromiseResolve = resolve
+            })
+            send()
+        } else if (statusDataPromise) {
+            statusDataPromise.then((data) => {
+                setData(data)
+            })
+        }
+    }, [send, wsData])
```

Comment:
There is some hack here. I do not understand all this ceremony. Moreover,
1. Your effect will not be regenerated if statusDataPromise or statusDataPromiseResolve changed
2. It is somehow dependent on the real-time code-flow, eg there is a chance that you call statusDataPromiseResolve(wsData) for a _null_

Seems like this block needs to be rewritten.

Reply:
here we are using promise and resolve to if two request going on a same time  we are detecting the in progress request and avoiding the new request. 
and for null we are checking if(wsData) so null values wont hit  statusDataPromiseResolve(wsData) this line